libs/wingc/src/capture.rs

```diff
@@ -246,10 +248,21 @@ fn scan_captures_in_expression(exp: &Expr, env: &SymbolEnv, statement_idx: usize
 							})
 							.collect::<Vec<Capture>>(),
 					);
+				} else if t.is_immutable_collection() || t.is_primitive() {
```

Comment:
Shouldn't this only capture stuff defined in preflight?
```suggestion
				} else if (t.is_immutable_collection() || t.is_primitive()) && f == Phase::PreFlight {
```